### PR TITLE
Feature/end to end tests

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-    "name": "symfony/framework-standard-edition",
+    "name": "transformCore/oneyou-haytool-persistence-api",
```
Comment (eddiejaoude) [inline composer.json]: `oneyou`?

File: phpunit.xml [hunk 2]
```diff
@@ -3,9 +3,10 @@
 <!-- http://phpunit.de/manual/4.1/en/appendixes.configuration.html -->
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.1/phpunit.xsd"
-         backupGlobals="false"
+         backupGlobals="true"
          colors="true"
          bootstrap="app/bootstrap.php.cache"
+         processIsolation="true"
```
Comment (eddiejaoude) [inline phpunit.xml]: Why? This causes other issues in the future and is dog slow.

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Address.php [hunk 3]
```diff
@@ -25,7 +25,7 @@ class Address implements IsIdentifiable
      * @ORM\Column(type = "json_array", name="address_lines", nullable = true)
      * @var array
      */
-    protected $addressLines = array();
+    protected $addressLines = '';
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Address.php]: `@var array`?

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Address.php [hunk 4]
```diff
@@ -63,7 +63,7 @@ public function getAddressLines()
      * @param array $addressLines
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Address.php]: `array`?

Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Entity/Address.php]: Good spot

File: phpunit.xml (hunk 2 above)
Comment (brettminnie) [inline phpunit.xml]: Need to figure out the why, doctrine errors out when this is set to false.
May be due to the way I am creating the db

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Tests/Repository/AddressRepositoryFunctionalTest.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    protected $result = null;
+
+    /** @var  AddressRepository */
+    protected $repository;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->repository = new AddressRepository($this->_em, new ClassMetadata(Address::class));
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Tests/Repository/AddressRepositoryFunctionalTest.php]: `_em` is not PSR*

File: composer.json (hunk 1 above)
Comment (brettminnie) [inline composer.json]: Global product name will address

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Tests/Repository/AddressRepositoryFunctionalTest.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+use TransformCore\PHE\HayApi\PersistenceBundle\Tests\RepositoryTestCase;
+
+/**
+ * Class AddressRepositoryFunctionalTests
+ * @package TransformCore\PHE\HayApi\PersistenceBundle\Tests\Repository
+ */
+class AddressRepositoryFunctionalTest extends RepositoryTestCase
+{
+    /** @var Address */
+    protected $result = null;
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Tests/Repository/AddressRepositoryFunctionalTest.php]: `null`?

File: src/TransformCore/PHE/HayApi/PersistenceBundle/Tests/Repository/AddressRepositoryFunctionalTest.php [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+    /** @var  AddressRepository */
+    protected $repository;
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Tests/Repository/AddressRepositoryFunctionalTest.php]: not `null`? Can it be consistent with above?

Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/PersistenceBundle/Tests/Repository/AddressRepositoryFunctionalTest.php]: For you of course :)

On Fri, Oct 30, 2015 at 5:48 PM, Eddie Abou-Jaoude <notifications@github.com

**Pull request merged**